specutils/manipulation/resample.py

```diff
@@ -157,64 +248,45 @@ def resample1d(self, orig_spectrum, fin_spec_axis):
         if not isinstance(fin_spec_axis, SpectralAxis):
             fin_spec_axis = SpectralAxis(fin_spec_axis)
 
-        # todo: Would be good to return uncertainty in type it was provided?
-        # todo: add in weighting options
-
         # Get provided uncertainty into variance
         if orig_spectrum.uncertainty is not None:
             pixel_uncer = orig_spectrum.uncertainty.represent_as(VarianceUncertainty).array
         else:
             pixel_uncer = None
 
+        # convert unit
         orig_axis_in_fin = orig_spectrum.spectral_axis.to(fin_spec_axis.unit)
-        resample_grid = self._resample_matrix(orig_axis_in_fin, fin_spec_axis)
-
-        # Now for some broadcasting magic to handle multi dimensional flux inputs
-        # Essentially this part is inserting length one dimensions as fillers
-        # For example, if we have a (5,6,10) input flux, and an output grid
-        # of 3, flux will be broadcast to (5,6,1,10) and resample_grid will
-        # Be broadcast to (1,1,3,10).  The sum then reduces down the 10, the
-        # original dispersion grid, leaving 3, the new dispersion grid, as
-        # the last index.
-        new_flux_shape = list(orig_spectrum.flux.shape)
-        new_flux_shape.insert(-1, 1)
-        in_flux = orig_spectrum.flux.reshape(new_flux_shape)
-
-        ones = [1] * len(orig_spectrum.flux.shape[:-1])
-        new_shape_resample_grid = ones + list(resample_grid.shape)
-        resample_grid = resample_grid.reshape(new_shape_resample_grid)
-
-        # Calculate final flux
-        out_flux = np.sum(in_flux * resample_grid, axis=-1) / np.sum(
-            resample_grid, axis=-1)
-
-        # Calculate output uncertainty
-        if pixel_uncer is not None:
-            pixel_uncer = pixel_uncer.reshape(new_flux_shape)
-
-            out_variance = np.sum(pixel_uncer * resample_grid**2, axis=-1) / np.sum(
-                resample_grid, axis=-1)**2
-            out_uncertainty = InverseVariance(np.reciprocal(out_variance))
+
+        # handle 2d flux inputs
+        if orig_spectrum.flux.ndim == 2:
```

Comment:
I think this is the only thing in this implementation that worries me - the previous one handled any higher dimensionality for flux, and I think that the most common cases are 1D and 3D. Could you generalize this at least to 3D?

Reply:
done!